### PR TITLE
Consolidate AOT Relocation Records

### PR DESCRIPTION
**Pull request by dsouzai**
https://github.com/eclipse/openj9/issues/4803

### CODE REVIEW

**Comment by dsouzai** [comment]
@mstoodle could you please review?

File: runtime/compiler/codegen/J9AheadOfTimeCompile.cpp [hunk 1]
```diff
@@ -240,6 +243,87 @@ J9::AheadOfTimeCompile::initializeCommonAOTRelocationHeader(TR::IteratedExternal
...
+
+         uint32_t branchOffset = static_cast<uint32_t>(label->getCodeLocation() - instr->getBinaryEncoding());
+
+         allocRecord->setInlinedSiteIndex(reloTarget, static_cast<uintptrj_t>(recordInfo->data2));
+         allocRecord->setConstantPool(reloTarget, reinterpret_cast<uintptrj_t>(classSymRef->getOwningMethod(comp)->constantPool()));
+         allocRecord->setBranchOffset(reloTarget, static_cast<uintptrj_t>(branchOffset));
+         allocRecord->setAllocationSize(reloTarget, static_cast<uintptrj_t>(recordInfo->data1));
+
+         /* Temporary, will be cleaned up in a future PR */
+         if (comp->getOption(TR_UseSymbolValidationManager))
```
Comment (mstoodle) [inline runtime/compiler/codegen/J9AheadOfTimeCompile.cpp]: This code wasn't consistently being applied on all platforms (e.g. on ARM there only a `cpIndex` was being written). I'm ok with making this change as part of the consolidation  as part of its "value", but please add a note in the appropriate commit messages that this change occurs.

**Comment by mstoodle** [comment]
Jenkins test sanity all jdk8,jdk11

**Comment by mstoodle** [comment]
since this covers all platforms, let's take extra care and do 'all'

**Comment by dsouzai** [comment]
lol
```
13:21:21  Downstream job Test_openjdk8_j9_sanity.functional_x86-32_windows_Personal PASSED after 1 hr 0 min
[Pipeline] node
13:21:21  Running on Jenkins in /home/hudson/genie.openj9/.jenkins/jobs/Pipeline_Build_Test_JDK8_x86-32_windows/workspace
[Pipeline] {
[Pipeline] step
13:21:24  [Set GitHub commit status (universal)] SUCCESS on repos [GHRepository@58b63527[description=Eclipse OpenJ9: A Java Virtual Machine for OpenJDK that's optimized for small footprint, fast start-up, and high throughput.   Builds on Eclipse OMR (https://github.com/eclipse/omr) and combines with the Extensions for OpenJDK for OpenJ9 repo.,homepage=,name=openj9,fork=false,archived=false,size=52696,milestones={},language=Java,commits={},source=<null>,parent=<null>,responseHeaderFields={null=[HTTP/1.1 200 OK], Access-Control-Allow-Origin=[*], Access-Control-Expose-Headers=[ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type], Cache-Control=[private, max-age=60, s-maxage=60], Content-Encoding=[gzip], Content-Security-Policy=[default-src 'none'], Content-Type=[application/json; charset=utf-8], Date=[Tue, 28 May 2019 17:21:24 GMT], ETag=[W/"0698152c4486d01dfd95b42704f23536"], Last-Modified=[Tue, 28 May 2019 17:18:46 GMT], OkHttp-Received-Millis=[1559064084533], OkHttp-Response-Source=[CONDITIONAL_CACHE 200], OkHttp-Selected-Protocol=[http/1.1], OkHttp-Sent-Millis=[1559064082123], Referrer-Policy=[origin-when-cross-origin, strict-origin-when-cross-origin], Server=[GitHub.com], Status=[200 OK], Strict-Transport-Security=[max-age=31536000; includeSubdomains; preload], Transfer-Encoding=[chunked], Vary=[Accept, Authorization, Cookie, X-GitHub-OTP], X-Accepted-OAuth-Scopes=[repo], X-Content-Type-Options=[nosniff], X-Frame-Options=[deny], X-GitHub-Media-Type=[github.v3; format=json], X-GitHub-Request-Id=[8D3C:192F:23492EB:4C2EC0B:5CED6E11], X-OAuth-Scopes=[repo:status, repo], X-RateLimit-Limit=[5000], X-RateLimit-Remaining=[4590], X-RateLimit-Reset=[1559067684], X-XSS-Protection=[1; mode=block]},url=https://api.github.com/repos/eclipse/openj9,id=101408117]] (sha:4ec4b63) with context:Test_openjdk8_j9_sanity.functional_x86-32_windows_Personal
13:21:24  Setting commit status on GitHub for https://github.com/eclipse/openj9/commit/4ec4b6329077cb673fe023c69d421ab1cbf52b36
13:21:24  ERROR: [GitHub Commit Status Setter] - {
13:21:24    "message": "Server Error"
13:21:24  }
13:21:24  , setting build result to UNSTABLE
```

**Comment by mstoodle** [comment]
The one windows 32-bit failure isn't really a failure, as @dsouzai pointed out, so looks good to merge.

Thanks @dsouzai for bringing us one step closer to consolidated relocation infrastructure across our code generators!

**Pull request merged**